src/main/java/com/google/devtools/build/lib/analysis/PlatformConfiguration.java

```diff
@@ -111,4 +114,23 @@ public Label getTargetPlatform() {
     }
     return constraints.build();
   }
+
+  /** Returns true if toolchain resolution debug info should be printed for this toolchain type. */
+  public boolean debugToolchainResolution(Label toolchainType) {
+    return debugToolchainResolution(ImmutableList.of(toolchainType));
+  }
+
+  /**
+   * Returns true if toolchain resolution debug info should be printed for any of these toolchain
+   * types.
+   */
+  public boolean debugToolchainResolution(Collection<Label> toolchainTypes) {
+    if (toolchainTypes.isEmpty()) {
+      // Check an empty string, in case the filter is .*
+      return this.toolchainResolutionDebugRegexFilter.test("");
```

Comment:
Does plain old `--toolchain_resolution_debug` still work? (as an alias for `.*`)?

Reply:
No, the argument is required (and if it is skipped an error will be printed). I couldn't figure out a way to distinguish between no flag being passed, and a base `--toolchain_resolution_debug` with no argument in order to keep the current behavior unchanged.